kite-morphlines/kite-morphlines-maxmind/src/test/java/org/kitesdk/morphline/maxmind/MaxmindMorphlineTest.java

```diff
@@ -15,75 +15,150 @@
+    }
+  }
+
   @Test
   public void testIPv4() throws Exception {
-    morphline = createMorphline("test-morphlines/geoIP");    
-    
-    Record record = new Record();
-    String ip = "128.101.101.101";
+    morphline = buildCommand("ip", "GeoLite2-City.mmdb");
+    final Record record = new Record();
+    final String ip = "128.101.101.101";
     record.put("ip", ip);
-    Record expected = new Record();
-    expected.put("ip", ip);
-    expected.put("/country/iso_code", "US");
-    expected.put("/country/names/en", "United States");
-    expected.put("/country/names/zh-CN", "美国");
-    expected.put("/subdivisions[]/names/en", "Minnesota");
-    expected.put("/subdivisions[]/iso_code", "MN");
-    expected.put("/city/names/en", "Minneapolis");
-    expected.put("/postal/code", "55455");
-    expected.put("/location/latitude", 44.9733);
-    expected.put("/location/longitude", -93.2323);
-    expected.put("/location/latitude_longitude", "44.9733,-93.2323");
-    expected.put("/location/longitude_latitude", "-93.2323,44.9733");
-    
-    processAndVerifySuccess(record, expected, false);
-    Notifications.notifyShutdown(morphline);
-    Notifications.notifyShutdown(morphline);
+    processAndVerifySuccess(record, true, false, 6252001, 5037649);
+  }
+
+  @Test
+  public void testIPv4CountryOnly() throws Exception {
+    morphline = buildCommand("ip", "GeoLite2-Country.mmdb");
+    final Record record = new Record();
+    final String ip = "128.101.101.101";
+    record.put("ip", ip);
+    processAndVerifySuccess(record, true, false, 6252001, null);
+  }
+
+  @Test
+  public void testNotFoundIPv4() throws Exception {
+    morphline = buildCommand("ip", "GeoLite2-City.mmdb");
+    final Record record = new Record();
+    final String ip = "127.0.0.1";
+    record.put("ip", ip);
+    processAndVerifySuccess(record, false, false, null, null);
+  }
+
+  @Test
+  public void testBadData() throws Exception {
+    morphline = buildCommand("ip", "GeoLite2-City.mmdb");
+    final Record record = new Record();
+    final String ip = "DAT DATA";
+    record.put("ip", ip);
+    processAndVerifySuccess(record, false, false, null, null);
+  }
+
+  @Test
+  public void testEmpty() throws Exception {
+    morphline = buildCommand("ip", "GeoLite2-City.mmdb");
+    final Record record = new Record();
+    processAndVerifySuccess(record, false, false, null, null);
   }
   
   @Test
   public void testIPv6() throws Exception {
-    morphline = createMorphline("test-morphlines/geoIP");    
-    
-    Record record = new Record();
-    String ip = "2001:620::1";
+    morphline = buildCommand("ip", "GeoLite2-City.mmdb");
+    final Record record = new Record();
+    final String ip = "2001:4860:4860::8844";
     record.put("ip", ip);
-    Record expected = new Record();
-    expected.put("ip", ip);
-    expected.put("/country/iso_code", "CH");
-    expected.put("/country/names/en", "Switzerland");
-    expected.put("/country/names/zh-CN", "瑞士");
```

Comment:
Should I keep 2 processAndVerifySuccess methods? One for the old tests and another for the new tests?